agents_infra/agents_infra/agents/base.py

```diff
@@ -810,3 +934,55 @@ def generate_report(self):
             'load_avg': self.get_load_average(),
             'timestamp': datetime.datetime.now().isoformat(),
         }
+
+    def send_metrics_to_controller(
+            self,
+            suffix='agent/metrics/',
```

Comment:
The name `suffix` is slightly ambiguous. Consider renaming to something more descriptive like `endpoint_path` or `metrics_endpoint` for better readability.